proc/tool/analysis/analysis.cpp

```diff
@@ -0,0 +1,238 @@
+     * -b <time_block>
+     * -d <directory>
+     * -h <help>
+     * -x <x_axis>
+     * -y <y_axis>
+     */
+    for (int i = 1; i < argc; i++) {
+        if (strcmp(argVector[i], "-s") == 0) {
+            startX = atoi(argVector[i + 1]);
+            endX = atoi(argVector[i + 2]);
+
+            i += 2;
+        } else if (strcmp(argVector[i], "-b") == 0) {
+            block = atoi(argVector[i + 1]);
+
+            i++;
+        } else if (strcmp(argVector[i], "-d") == 0) {
+            if (i + 1 >= argc) {
+                cout << "Error: -d requires a directory path" << endl;
+                return false;
+            } else {
+                directory = argVector[i + 1];
+                directorySet = true;
+            }
+
+            i++;
+        } else if (strcmp(argVector[i], "-x") == 0) {
+            if (i + 1 >= argc) {
+                cout << "Error: -x requires an x variable" << endl;
+            } else {
+                xSet = true;
+                xAxis = argVector[i + 1];
+            }
+
+            i++;
+        } else if (strcmp(argVector[i], "-y") == 0) {
+            if (i + 1 >= argc) {
+                cout << "Error: -y requires a y variable" << endl;
+            } else {
+                ySet = true;
+                yAxis = argVector[i + 1];
+            }
+
+            i++;
+        } else if (strcmp(argVector[i], "-h") == 0) {
+            cout << "Usage: ./analysis -s <start_x> <end_x> -b <time_block> -d <directory>" << endl;
+            helpSet = true;
+            break;
+        }
+    }
+
+    if (!(!helpSet && directorySet && xSet && ySet)) {
+        cout << "Error: Missing required arguments" << endl;
+        return false;
+    }
+
+    return true;
+
+}
+
+
+vector<string> splitCSVLine(const string &str) {
+    vector<string> result;
+
+    stringstream ss(str);
+    string token;
+
+    while (getline(ss, token, ',')) {
+        result.push_back(token);
+    }
+    return result;
+}
+
+array<vector<npy_double>, 2> parseCSV(const string &filename) {
+    int millisecColNum = -1;
+    int microsecColNum = -1;
+    int xColNum = -1;
+    int yColNum = -1;
+
+    if (!fs::exists(filename)) {
+        cout << "File does not exist" << endl;
+        return {};
+    }
+
+    ifstream file(filename);
+    vector<npy_double> xVals;
+    vector<npy_double> yVals;
+
+    string line;
+
+    getline(file, line);
+    vector<string> headers = splitCSVLine(line);
+
+    for (int i = 0; i < headers.size(); i++) {
+        if (headers[i] == "MILLISEC") {
+            millisecColNum = i;
+        }
+
+        if (headers[i] == "MICROSEC") {
+            microsecColNum = i;
+        }
+
+        if (headers[i] == xAxis) {
+            xColNum = i;
+        }
+
+        if (headers[i] == yAxis) {
+            yColNum = i;
+        }
+    }
+
+    if (millisecColNum == -1 || microsecColNum == -1 || xColNum == -1 || yColNum == -1) {
+        cout << "Columns not found" << endl;
+        return {};
```

Comment:
should probably close file here as well, probably fine if the process just exits after this since the file will be closed automatically